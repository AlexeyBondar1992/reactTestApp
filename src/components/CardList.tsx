import React from 'react';
import Card from './Card';
import {IRobot} from '../interfaces';


const CardList: React.FunctionComponent<{ robots: IRobot[] }> = ({ robots }) => {
    return (
        <div>
            {
                robots.map(robot=> {
                    return <Card key={robot.id}
                                 id={robot.id}
                                 name={robot.name}
                                 email={robot.email}
                    />;
                })
            }
        </div>);
};

export default CardList;