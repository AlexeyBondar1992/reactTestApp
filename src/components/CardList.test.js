import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import CardList from './CardList';

const mockRobots = [
    {
        id: 1,
        name: 'John Snow',
        userName: 'JohnJohn',
        email: 'john@gmail.com'
    }
];

describe('CardList', () => {
    it('should test component creation', () => {
        expect(shallow(<CardList robots={mockRobots}/>).length).toEqual(1);
    });

    it('should test snapshot to match', () => {
        expect(toJson(shallow(<CardList robots={mockRobots}/>))).toMatchSnapshot();
    });
});
