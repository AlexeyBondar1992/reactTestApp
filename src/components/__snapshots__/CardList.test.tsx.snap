// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList should test snapshot to match 1`] = `
<div>
  <Card
    email="john@gmail.com"
    id={1}
    key="1"
    name="John Snow"
  />
</div>
`;
